src/lib/core/fiber.h

```diff
@@ -842,6 +848,42 @@ fiber_time_from_call(struct fiber *f)
 	return clock_monotonic64() - f->call_time;
 }
 
+/**
+ * Check if deadline is up for current cord.
+ */
+static inline bool
+check_deadline()
+{
+	return cord()->current_fiber_deadline_timeout > fiber_time_from_call(fiber());
+}
+
+/**
+ * Set default deadline timeout for new fibers of current cord.
+ */
+void
+set_default_deadline_timeout(uint64_t new_deadline_timeout);
+
+/**
+ * Set deadline timeout to fiber. Must be called in the thread in which the cord
+ * that manages this fiber is located.
+ */
+void
+fiber_set_deadline_timeout(struct fiber *fiber, uint64_t new_deadline_timeout);
+
+/**
+ * Increase current deadline timeout of current cord. Must be called in the thread
+ * in which the cord that manages this fiber is located.
+ */
+void
+increase_deadline_timeout(uint64_t time_added);
+
+/**
+ * Set current deadline timeout of current cord to zero which means that deadline
+ * for current fiber execution is up.
+ */
+void
+force_deadline();
```

Comment:
All these functions could be inline - they are one-liners.

Reply:
fixed